datastore/metadata.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+var gcloud = require('gcloud');
+
+// This mock is used in the documentation snippets.
+var datastore = {
+  runQuery: function() {}
+};
+
+function Metadata(projectId) {
+  this.datastore = gcloud.datastore({
+    projectId: projectId
+  });
+}
+
+Metadata.prototype.testNamespaceRunQuery = function(callback) {
+  var self = this;
+
+  datastore.createQuery = this.datastore.createQuery;
+  datastore.key = this.datastore.key;
+
+  var startNamespace = 'Animals';
+  var endNamespace = 'Zoos';
+
+  this.datastore.save([
+    {
+      key: datastore.key({
+        namespace: 'Animals',
+        path: ['Ant', 1]
+      }),
+      data: {}
+    }
+  ], function(err) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    // jshint unused:false
+    // [START namespace_run_query]
+    var query = datastore.createQuery('__namespace__')
+      .select('__key__')
+      .filter('__key__ >=', datastore.key(['__namespace__', startNamespace]))
+      .filter('__key__ <', datastore.key(['__namespace__', endNamespace]));
+
+    datastore.runQuery(query, function(err, entities) {
+      if (err) {
+        // An error occurred while running the query.
+        return;
+      }
+
+      var namespaces = entities.map(function(entity) {
+        return entity.key.path.pop();
+      });
+    });
+    // [END namespace_run_query]
+
+    self.datastore.runQuery(query, callback);
+  });
+};
+
+Metadata.prototype.testKindRunQuery = function(callback) {
+  datastore.createQuery = this.datastore.createQuery;
+
+  // jshint unused:false
+  // [START kind_run_query]
+  var query = datastore.createQuery('__kind__')
+    .select('__key__');
+
+  datastore.runQuery(query, function(err, entities) {
+    if (err) {
+      // An error occurred while running the query.
+      return;
+    }
+
+    var kinds = entities.map(function(entity) {
+      return entity.key.path.pop();
+    });
+  });
+  // [END kind_run_query]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Metadata.prototype.testPropertyRunQuery = function(callback) {
+  datastore.createQuery = this.datastore.createQuery;
+
+  // [START property_run_query]
+  var query = datastore.createQuery('__property__')
+    .select('__key__');
+
+  datastore.runQuery(query, function(err, entities) {
+    if (err) {
+      // An error occurred while running the query.
+      return;
+    }
+
+    var propertiesByKind = {};
+
+    entities.forEach(function(entity) {
+      var kind = entity.key.path[1];
+      var propertyName = entity.key.path[3];
+
+      propertiesByKind[kind] = propertiesByKind[kind] || [];
+      propertiesByKind[kind].push(propertyName);
+    });
+
+    // propertiesByKind = {
```

Comment:
We should build (and the other comments) in a language-agnostic way into the docs around the snippets, so no need to include them here.